datadog/fwprovider/resource_datadog_apm_retention_filter.go

```diff
@@ -132,12 +133,73 @@ func (r *ApmRetentionFilterResource) Read(ctx context.Context, request resource.
 	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
 }
 
+type CommonRequest struct {
+	Diagnostics diag.Diagnostics
+	State       *tfsdk.State
+}
+
+func NewCommonRequest(diag diag.Diagnostics, state *tfsdk.State) CommonRequest {
+	return CommonRequest{
+		Diagnostics: diag,
+		State:       state,
+	}
+}
+
+func (r *ApmRetentionFilterResource) getAndUpdate(state *ApmRetentionFilterModel, ctx context.Context, response CommonRequest) {
+	resp, _, err := r.Api.ListApmRetentionFilters(r.Auth)
+	if err != nil {
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		return
+	}
+	var id string
+	var filterName string
+	for _, rfa := range resp.Data {
+		if string(rfa.Attributes.GetFilterType()) == state.FilterType.ValueString() {
```

Comment:
Why are we matching my filter type here? Can't we have multiple default pipelines with the same type?